### PR TITLE
use ZTX_LOG in more places, early returns on service update checking

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- replaces ZITI_LOG  calls with ZTX_LOG where possible
- service update checking will now return earlier rather than performing
  additional checks

### CODE REVIEW

File: includes/ziti/ziti_log.h [hunk 1]
```diff
@@ -43,6 +43,8 @@ enum DebugLevel {
 if (level <= ziti_log_level()) { ziti_logger(level, __FILE__, __LINE__, __func__, fmt, ##__VA_ARGS__); }\
 } while(0)
 
+#define ZTX_LOG(lvl, fmt, ...) ZITI_LOG(lvl, "ztx[%d] " fmt, ztx->id, ##__VA_ARGS__)
```
Comment (dovholuknf) [inline includes/ziti/ziti_log.h:46]: do we really want this exposed to consumers?

File: library/auth_queries.c [hunk 2]
```diff
@@ -159,29 +159,31 @@ void ziti_auth_query_process(ziti_context ztx, void(*cb)(ziti_context, int, void
...
-        ZITI_LOG(ERROR, "error during create MFA call: %d - %s - %s", err->http_code, err->code, err->message);
+        ZTX_LOG(ERROR, "error during create MFA call: %d - %s - %s", err->http_code, err->code, err->message);
         mfa_enroll_cb_ctx->cb(mfa_enroll_cb_ctx->ztx, err->err, NULL, mfa_enroll_cb_ctx->cb_ctx);
         FREE(ctx);
     }
 }
 
 void ziti_mfa_enroll_get_internal_cb(ziti_mfa_enrollment *mfa_enrollment, const ziti_error *err, void *ctx) {
     ziti_mfa_enroll_cb_ctx *mfa_enroll_cb_ctx = ctx;
+    ziti_context ztx = mfa_enroll_cb_ctx->ztx;
```
Comment (dovholuknf) [inline library/auth_queries.c:182]: same comment here - is ztx used somehow?

File: library/auth_queries.c [hunk 3]
```diff
@@ -159,29 +159,31 @@ void ziti_auth_query_process(ziti_context ztx, void(*cb)(ziti_context, int, void
...
 
-    ZITI_LOG(INFO, "sending auth query mfa authentication event, authentication cannot proceed if not handled");
+    ZTX_LOG(INFO, "sending auth query mfa authentication event, authentication cannot proceed if not handled");
     ziti_auth_query_mfa_process(mfa_auth_ctx);
 }
 
 
 void ziti_mfa_enroll_post_internal_cb(void *empty, const ziti_error *err, void *ctx) {
     ziti_mfa_enroll_cb_ctx *mfa_enroll_cb_ctx = ctx;
+    ziti_context ztx = mfa_enroll_cb_ctx->ztx;
```
[the remark this answers is not in the document]
Comment (andrewpmartinez) [reply library/auth_queries.c:169]: `ZTX_LOG` macro expects a scope defined `ztx`

File: library/auth_queries.c (hunk 2 above)
Comment (andrewpmartinez) [reply library/auth_queries.c:182]: ZTX_LOG macro expects a scope defined ztx

File: library/auth_queries.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (dovholuknf) [reply library/auth_queries.c:169]: ah. well then that makes a lot more sense :) thx

File: includes/ziti/ziti_log.h (hunk 1 above)
Comment (andrewpmartinez) [reply includes/ziti/ziti_log.h:46]: Ahh hmm. I don't know what would happen if they tried to use it. 

I can move it to `zt_internal.h`. Is that a better place for it @ekoby ?

Comment (ekoby) [reply includes/ziti/ziti_log.h:46]: yes

File: programs/ziti-prox-c/proxy.c [hunk 4]
```diff
@@ -479,6 +479,24 @@ static void on_ziti_event(ziti_context ztx, const ziti_event_t *event) {
                     service_check_cb(ztx, *sp, ZITI_OK, app_ctx);
                 }
             }
+
+            if (event->event.service.changed != NULL) {
```
Comment (andrewpmartinez) [inline programs/ziti-prox-c/proxy.c:483]: Allows for mfa prompting on posture check timeouts.

**Review by dovholuknf — APPROVED** [review]

**Pull request merged**